### DISCUSSION TITLE
"Files changed" view in pull request not loading files and comments on large PR

### DISCUSSION THREAD

**Discussion by matteoolivi** [community/community#11122] [Pull Requests]
I have a PR that is really large: 17 files changed, 2.7k lines of code added, 0 removed, 190 review comments.

When I click in the "files changed" view to add comments or reply to existing ones, the view loads only 2 of the 17 files (and of course, it doesn't show the comments on those 15 files that are not displayed):

![image](https://user-images.githubusercontent.com/72204904/152778556-21000433-1a19-4d9d-a993-5f08db2f59aa.png)

It hangs indefinitely, or crashes after some time. Other PRs against the same repo, of normal size, are still working perfectly well (the "files changed" view displays everything).

This happens for Chrome, Safari and Firefox, and is happening to all of my colleagues who are trying to look at the PR, not just me (so it's not due to my machine doing something weird).

I'm not sure what's causing the issue exactly, but before there was such a big number of comments the "files changed" view was working perfectly fine.

I know, PRs this big are an anti-pattern. Yet, I feel like github should work well even in this case. Can something be done about it? How can I find out more precisely what's causing the issue?

**Comment by matteoolivi** [comment]
cc @mkfdoherty

**Comment by theetrain** [comment]
I also face this issue in pull requests with 50 files. Most of those files only have a handful or a few dozen lines of code changes in their diffs. Sometimes the page loads fine, but after submitting 20 comments in a PR review, my colleague would push more commits and then the `/files` page hangs, similarly to what was described by OP.

On a 2015 Macbook pro with 16GB RAM running Chrome 98, my page crashes when loading `/files` in a PR:
![image](https://user-images.githubusercontent.com/12798751/154750823-840d3c79-f639-4611-9861-44464861cc22.png)

On a custom PC running Windows 10 with 16 GB RAM, it also crashes but with more details related to memory::
![Image 2022-02-18 at 2 42 59 PM](https://user-images.githubusercontent.com/12798751/154751031-c2461bae-3661-4447-98a6-370bbf57c200.jpg)


The page _does_ work on a 2021 Macbook Pro M1 Max with 64 GB ram, but I shouldn't need 64 GB ram.

Of course I agree pull requests should be small, but sometimes they go through journeys with multiple comments spanning complex commit diffs that are outdated, and the journey to go back and resolve older comments should be possible, such as utilizing the `Conversations` tab at the top of the page. It's great that GitHub continues to improve the tools in PR reviews, and I hope this performance issue can be addressed for large-scale PRs.

**Reply by willsmythe** [reply]
Can you try viewing this PR again? We fixed a bug last week that should resolve this.

**Reply by theetrain** [reply]
Performance seems to have improved, thank you.

**Comment by billxinli** [comment]
I am getting page is unresponsive with large PRs. 

I am running Chrome on a 5950x with 128GB of RAM.

![Screenshot from 2022-02-04 10-52-30](https://user-images.githubusercontent.com/1641464/154754956-103ef508-5b30-4d50-86bf-fd318bd79224.png)

I am happy to provide the link to the PR from above to GH if required.

**Reply by willsmythe** [reply]
Can you try viewing this PR again? We fixed a bug last week that should resolve this.

**Comment by FaithOmbongi** [comment]
Same situation here.
I've tested on both Chrome and Edge and in Incognito mode, but the freeze is prevalent. The PR is 21 files long.

<img width="821" alt="image" src="https://user-images.githubusercontent.com/14026935/155355264-dd175b33-f979-451a-92f3-022f49b0f0d6.png">

**Comment by ravenshining** [comment]
I found my way here because of the same issue, but I deal with much larger PRs.  ~135 files changed, +1500 lines & -500 lines one I've got open right now, and that's the *small slice of something five times larger so it's easier to review* version.  I wanted to break it up even smaller, but was told not to as that would create too many PRs to sort through.

It'd be really nice if it didn't auto-expand everything to begin with on a large PR, and refrained from loading things until you click to expand them in the left pane.  Trying to load *everything* at once hangs the browser and eats data unnecessarily.

**Comment by tanchekwei** [comment]
I'm experiencing the same issue. Unresponsive and unable to navigate the file tree for the pull request:
https://github.com/files-community/Files/pull/10283/files

**Comment by grasmanek94** [comment]
Hi I dropped here from a search engine during my journey in search of a GitHub website PR/branch diff compare. As it stands now, Files changed views for PR's are loaded into a single view/page. I would like to suggest an option to be able to just load a single file per page (that file being the selected file in the sidebar) instead of all files. Could this be added, please?
